### PR TITLE
Menu & hand controller polish

### PR DESCRIPTION
**Pull request by jherico**
* Make the highlight for QML menus instantly move to the reticle position, rather than lagging behind it and animating.
* Make the trigger 'click' on the Vive be the signal to trigger grab or mouse click behavior from `handControllerGrab.js` and `handControllerPointer.js`.  For Oculus Touch, use a cutoff value of 0.9.

## Testing notes

With the Vive controller, neither the grab or mouse click effect should occur until the trigger is fully depressed to the extent that it 'clicks'.  In the production build, the grab or mouse pointer can occur prior to this point.  

When opening a context menu on the QML 2D layer, the highlight should follow the mouse cursor position precisely, rather than animating over time to the current location.  Note that there may still be a small delay between the laser termination point moving and the pointer position updating due to threading and rendering issues, however the delay should be much shorter than the original animation delay.

### CODE REVIEW

**Comment by jherico** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;h56JHssh&#x2F;62-menu-polish"> Menu Polish</a>

**Comment by hifi-gustavo** [comment]
Build console output for commit: 808514433a6c6e2f1f59467d78a4ba9035613812 available at: https://gist.github.com/c99164ea765ad1bbfdafde21675c8f29<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8085144.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8085144.dmg

**Comment by RyanDowne** [comment]
What's the test plan on this?

**Comment by RyanDowne** [comment]
I tested this and I really like the new behavior. The only issue I see is that it's still possible to open the sub-menus without RT_CLICK. The best way I have been able to repro is to open the menu start the laser, move it off of the menu and back on to an item that has a nested menu. When you move if back on (or maybe it's when you move it off) the sub-menu appears. I would expect that the sub-menu doesn't appear until you have it highlit and get to 1.0.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8685b236101b20a44403710364ee7de10b992f06 available at: https://gist.github.com/c99164ea765ad1bbfdafde21675c8f29<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8685b23.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8685b23.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8d5b60a266482f740251def186a6c996946fef5c available at: https://gist.github.com/c99164ea765ad1bbfdafde21675c8f29<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8d5b60a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8d5b60a.dmg

**Comment by jherico** [comment]
test this please

**Comment by RyanDowne** [comment]
Test plan:

Using the Vive hand controllers

1) Press the trigger down a little. You should see the blue laser. You should not see the red laser until you pull the trigger all the way and hear the "click"
2) Open the menu. Press the trigger down until you see the blue laser. Move the blue laser over the menu items. It should not open the sub-menus when the laser is Blue. 
3) Full press the trigger on a menu item. It should open the nested menu. Full press the trigger on a menu option, it should select that option. 
4) Press the trigger down and move over the HUD buttons. They should not select until you fully press and hear the click. 
5) Open the Marketplace. Press the trigger down until you see the blue laser. Move it over the windows border. You should not be able to move the window around until you fully press the trigger and hear the click.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8d5b60a266482f740251def186a6c996946fef5c available at: https://gist.github.com/c99164ea765ad1bbfdafde21675c8f29<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8d5b60a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-8d5b60a.dmg

**Comment by thoys** [comment]
## QA Results

1. Press the trigger down a little. You should see the blue laser. You should not see the red laser until you pull the trigger all the way and hear the "click" :white_check_mark: 
2. Open the menu. Press the trigger down until you see the blue laser. Move the blue laser over the menu items. It should not open the sub-menus when the laser is Blue.  :white_check_mark:  
3. Full press the trigger on a menu item. It should open the nested menu. Full press the trigger on a menu option, it should select that option.  :white_check_mark: 
4. Press the trigger down and move over the HUD buttons. They should not select until you fully press and hear the click. :white_check_mark: 
5. Open the Marketplace. Press the trigger down until you see the blue laser. Move it over the windows border. You should not be able to move the window around until you fully press the trigger and hear the click. :white_check_mark: 

Bugs:
1. Found a small bug, where the selection beam appears to have moved into the far distance when you point it at certain directions in the sky:
image (NON QA breaking)

2. Also weren't the buttons supposed to light or move up once you hovered over them?

3. I was able to hover over a toolbar through a settings window.  The selection beam shouldn't make it through the QML/QT window.

Those 3 points do not fail this QA, but please review them and add them as bugs if they are not yet fixed and if they actually not happen to be features


QA Pass

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 1]
```diff
@@ -284,7 +288,9 @@ void ViveControllerManager::InputDevice::handleHandController(float deltaTime, u
 
         vr::VRControllerState_t controllerState = vr::VRControllerState_t();
         if (_system->GetControllerState(deviceIndex, &controllerState)) {
-
+            //std::stringstream stream;
+            //stream << std::hex << controllerState.ulButtonPressed << " " << std::hex << controllerState.ulButtonTouched;
+            //qDebug() << deviceIndex << " " << stream.str().c_str() << controllerState.rAxis[1].x;
```
Comment (SamGondelman) [inline plugins/openvr/src/ViveControllerManager.cpp]: debug code?

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 #include "ViveControllerManager.h"
 
+#include <sstream>
+#include <QtCore/QDebug>
```
Comment (SamGondelman) [inline plugins/openvr/src/ViveControllerManager.cpp]: unused includes for commented out debug code

File: scripts/system/controllers/handControllerPointer.js [hunk 3]
```diff
@@ -493,7 +495,10 @@ function checkSettings() {
     updateRecommendedArea();
 }
 checkSettings();
+
 var settingsChecker = Script.setInterval(checkSettings, SETTINGS_CHANGE_RECHECK_INTERVAL);
 Script.scriptEnding.connect(function () {
     Script.clearInterval(settingsChecker);
+    OffscreenFlags.navigationFocusDisabled = false;
```
Comment (SamGondelman) [inline scripts/system/controllers/handControllerPointer.js]: will this reenable the navigation focus regardless of if other scripts have disabled it?

File: plugins/openvr/src/ViveControllerManager.cpp (hunk 1 above)
Comment (jherico) [inline plugins/openvr/src/ViveControllerManager.cpp]: removed

File: plugins/openvr/src/ViveControllerManager.cpp (hunk 2 above)
Comment (jherico) [inline plugins/openvr/src/ViveControllerManager.cpp]: removed

File: scripts/system/controllers/handControllerPointer.js (hunk 3 above)
Comment (jherico) [inline scripts/system/controllers/handControllerPointer.js]: It will, but this flag is basically for the exclusive use of a script that is emulating a pointer, of which presumably only 1 is running at a time.  We don't really have a good solution for dealing with multiple scripts concurrently modifying global state, and solving that problem is out of scope for this PR.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 90299a98fec556ffc236b8eecb9c859b394887ed available at: https://gist.github.com/c99164ea765ad1bbfdafde21675c8f29<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-90299a9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8240-90299a9.dmg
